src/clean_air/visualise/dataset_renderer.py

```diff
@@ -26,74 +29,87 @@ def __init__(self, dataset):
         self.x_coord = self.y_coord = self.z_coord = self.t_coord = None
-        for i, cube in enumerate(self.plot_list):
-            if i == 0:
-                for coord in cube.coords():
-                    if len(coord.points) > 1:
-                        axis = iris.util.guess_coord_axis(coord)
-                        if axis == 'X' and self.x_coord is None:
-                            self.x_coord = coord.name()
-                        elif axis == 'Y' and self.y_coord is None:
-                            self.y_coord = coord.name()
-                        elif axis == 'Z' and self.z_coord is None:
-                            self.z_coord = coord.name()
-                        elif axis == 'T' and self.t_coord is None:
-                            self.t_coord = coord.name()
+        for coord in self.dims:
+            if len(coord.points) > 1:
+                axis = iris.util.guess_coord_axis(coord)
+                if axis == 'X' and self.x_coord is None:
+                    self.x_coord = coord.name()
+                elif axis == 'Y' and self.y_coord is None:
+                    self.y_coord = coord.name()
+                elif axis == 'Z' and self.z_coord is None:
+                    self.z_coord = coord.name()
+                elif axis == 'T' and self.t_coord is None:
+                    self.t_coord = coord.name()
             else:
                 pass
 
     def render(self):
         """
-        Analyses the dimensionality of the dataset and then sends to
-        appropriate renderer in render_plot.py or render_map.py.
+        Analyses the dimensionality of the dataset and then converts to appropriate pandas dataframe type.
         """
         coords = (self.x_coord, self.y_coord, self.z_coord, self.t_coord)
+        self.rendered_df = None
 
         # If we have both an x-coord and y-coord then we can draw a map:
         if self.x_coord is not None and self.y_coord is not None:
             self.img_type = 'map'
-            fig = render_map.Map(self.dataset).render(*coords)
+            # NOTE: Even if multiple cubes are passed in here, only the first cube will be converted to pandas to plot
+            # as a map.  Multiple maps cannot be plotted at once as only a single dataframe is returned.
+            self.rendered_df = iris.pandas.as_data_frame(self.plot_list[0])
+
         # If we have just a time coord then we can make a timeseries:
-        elif self.x_coord is None and self.y_coord is None and \
-                self.t_coord is not None:
+        elif self.x_coord is None and self.y_coord is None and self.t_coord is not None:
             self.img_type = 'timeseries'
-            fig = render_plot.Plot(self.plot_list).render_timeseries()
-        # If we don't have any coords then something's gone wrong and we can't
-        # plot anything:
+            if len(self.plot_list) == 1:        # This is a single cube in a CubeList, so convert to series:
```

Comment:
This has broken the plotting for the Timeseries notebooks, as hvplot treats seems to Series and Dataframes differently. We can either revert this to return Dataframes for timeseries, or try to rewrite the notebooks to work with Series.

Reply:
@corinnebosley This comment still stands - are you able to rewrite it to always return a dataframe or do you want me to tackle it when I merge everything?

Reply:
The problem was dataframes wouldn't load for timeseries, pandas is a real pain about whether it can load a series or a dataframe; that's why this fix went in.  However, this is a problem so I will write a ticket outlining the issue vaguely and we can add detail at a later date.  This may be more of an issue for the notebooks than the code itself, that remains to be seen.

Reply:
Actually, that may not be true.  Let me just reorganise some bits and see if I can make this bit work...

Reply:
Looks like I can just revert it, which makes me nervous because there was definitely an error about pandas creating a dataframe with a single cube before, but now it seems to have mysteriously vanished...

I'm sure it'll be fine.  Change is reverted, running tests now, should be a PR up soon.